packages/keystone/crypto/EncryptionManager.spec.js

```
@@ -0,0 +1,202 @@
+        const exampleString = faker.random.alphaNumeric(10)
+        const encryptedInDifferentVersionsStrings = []
+        const versions = {}
+
+        for (let i = 0; i < differentVersionsCount; i++) {
+            const versionsSingle = generateVersionsInMode('cbc', 1)
+            const encryptionVersionId = Object.keys(versionsSingle)[0]
+            const manager = new EncryptionManager({ versions: versionsSingle, encryptionVersionId  })
+            encryptedInDifferentVersionsStrings.push(manager.encrypt(exampleString))
+            versions[encryptionVersionId] = versionsSingle[encryptionVersionId]
+        }
+        expect(Object.keys(versions)).toHaveLength(differentVersionsCount)
+        expect(new Set(encryptedInDifferentVersionsStrings).size).toBe(differentVersionsCount)
+        const encryptionVersionId = faker.helpers.arrayElement(Object.keys(versions))
+        const manager = new EncryptionManager({ versions, encryptionVersionId })
+
+        encryptedInDifferentVersionsStrings.forEach(encryptedString => {
+            const decrypted = manager.decrypt(encryptedString)
+            expect(decrypted).toEqual(exampleString)
+        })
+    })
+    
+    describe('Unsuccessful decryption', () => {
+        const manager = new EncryptionManager({ 
+            versions: {
+                '1': { algorithm: 'aes-256-cbc', secret: faker.random.alphaNumeric(32) },
+            },
+            encryptionVersionId: '1',
+        })
+        
+        const nonStrings = [
+            [],
+            {},
+            null,
+            undefined,
+            new Date(),
+            123,
+            Symbol(),
+        ]
+        
+        test.each(nonStrings)('Passed non string %p', (nonString) => {
+            expect(manager.decrypt(nonString)).toBeNull()
+        })
+
+        test('Is not encrypted', () => {
+            const notEncrypted = faker.random.alphaNumeric(10)
+            expect(manager.decrypt(notEncrypted)).toBeNull()
```

it should throw an error ...